### DISCUSSION TITLE
API: Allow public view of individual item, but not list of all items

### DISCUSSION THREAD

**Discussion by geeksareforlife** [directus/directus#15188] [Archive]
Apologies if this has already been answered, I have tried searching but I am new to Directus and probably don't have the terminology correct!

I am trying to allow the Public role to view an individual item from a collection via the API, but not see the full list of items in the collection - is this possible?

The scenario I am looking to implement is that an authenticated user would be able to share a public link to their item, which anyone could see, but unauthenticated people can't see the full list.

thanks

GR4L

**Comment by JonathanSchndr** [comment]
For me, it sounds like you have to create your own custom endpoint: https://docs.directus.io/extensions/endpoints.html

**Reply by licitdev** [reply]
Yes, a custom endpoint can implement the stated behaviour. You may upvote https://github.com/directus/directus/discussions/11464 to support the feature request. 👍

**Reply by geeksareforlife** [reply]
Thank you both!  Endpoints look interesting, so I'll get reading the docs.  I have also upvoted that feature request!
